src/NuGet.Services.EndToEnd/ReadmeTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using NuGet.Services.EndToEnd.Support;
+using NuGet.Services.EndToEnd.Support.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.EndToEnd
+{
+    [Collection(nameof(PushedPackagesCollection))]
+    public class ReadmeTests
+    {
+        private readonly PushedPackagesFixture _pushedPackages;
+        private readonly Clients _clients;
+        private readonly ITestOutputHelper _logger;
+
+        public ReadmeTests(PushedPackagesFixture pushedPackages, ITestOutputHelper logger)
+        {
+            _pushedPackages = pushedPackages;
+            _clients = pushedPackages.Clients;
+            _logger = logger;
+        }
+
+        [Fact]
+        public async Task PushedPackageWithReadmeIsAvailableInRegistration()
+        {
+            // Arrange
+            var package = await _pushedPackages.PrepareAsync(PackageType.EmbeddedReadmeFile, _logger);
+            var expectedPath = GetExpectedReadmePath(package);
+
+            // Act & Assert
+            var packageRegistrationList = await _clients.Registration.WaitForPackageAsync(package.Id, package.FullVersion, semVer2: false, logger: _logger);

```

Comment:
>false [](start = 132, length = 5)

we should test both SemVer 1.0 and SemVer 2.0 registration hives

Reply:
It looks like setting the `semVer2` parameter to `false` is the right thing to do actually. A value of `true` excludes the legacy hives. See:

https://github.com/NuGet/NuGet.Services.EndToEnd/blob/68f53566e117a99bde9c92ed882e8d4c712a02dc/src/NuGet.Services.EndToEnd/Support/Clients/RegistrationClient.cs#L136

https://github.com/NuGet/NuGet.Services.EndToEnd/blob/68f53566e117a99bde9c92ed882e8d4c712a02dc/src/NuGet.Services.EndToEnd/Support/Clients/V3IndexClient.cs#L51-L55

https://github.com/NuGet/NuGet.Services.EndToEnd/blob/68f53566e117a99bde9c92ed882e8d4c712a02dc/src/NuGet.Services.EndToEnd/Support/Clients/V3IndexClient.cs#L87-L92

Reply:
FYI @lyndaidaii, we have multiple copies of the registration API:

1. The legacy hive - This used by older clients that don't support modern features
1. The gzipped hive - This is for NuGet client versions 3.4.0 to 3.6.0. The JSON is compressed to improve performance.
1. The semver2 hive - This is for NuGet client versions 3.6.0 and newer. This includes SemVer 2.0.0 packages (package with versions like `1.0.0-prerelease+metadata`) that aren't supported by older clients.

We want to make sure that your package exists in all of these hives. If your package had a semver2 version, you wouldn't want to wait on the legacy and gzipped hives as the package weill never show up there. For more information on this, see our docs: https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource

Reply:
I decided to to rename value here since if this is little confusing